crypto/provider_core.c

```diff
@@ -483,6 +483,7 @@ void ossl_provider_free(OSSL_PROVIDER *prov)
 int ossl_provider_set_module_path(OSSL_PROVIDER *prov, const char *module_path)
 {
     OPENSSL_free(prov->path);
+    prov->path = NULL;
```

Comment:
Rather than doing this would it be better to move the `module_path == NULL` check to before the `OPENSSL_free` call above?

Reply:
IMO not, the call can be used to clear the path this way.